file-formats/aifp/type/zif_aff_aifp_v1.intf.abap

```diff
@@ -0,0 +1,293 @@
+    BEGIN OF co_simple_field_check,
+      "! <p class="shorttext">Empty</p>
+      "! Empty
+      empty              TYPE ty_simple_field_check VALUE 'I',
+      "! <p class="shorttext">Not Empty</p>
+      "! Not empty
+      not_empty          TYPE ty_simple_field_check VALUE 'N',
+      "! <p class="shorttext">Empty or '0'</p>
+      "! Empty or '0'
+      empty_or_zero      TYPE ty_simple_field_check VALUE '0',
+      "! <p class="shorttext">Numeric (Integer)</p>
+      "! Numeric (Integer)
+      numeric_integer    TYPE ty_simple_field_check VALUE '1',
+      "! <p class="shorttext">Numeric or Empty</p>
+      "! Numeric or empty
+      numeric_empty      TYPE ty_simple_field_check VALUE '2',
+      "! <p class="shorttext">Numeric (Integer/Negative)</p>
+      "! Numeric (Integer/Negative)
+      numeric_negativ    TYPE ty_simple_field_check VALUE '3',
+      "! <p class="shorttext">Numeric (Comma/Negative)</p>
+      "! Numeric (Comma/Negative)
+      numeric_comma      TYPE ty_simple_field_check VALUE '4',
+      "! <p class="shorttext">Numeric (Dot/Negative)</p>
+      "! Numeric (Dot/Negative)
+      numeric_dot        TYPE ty_simple_field_check VALUE '5',
+      "! <p class="shorttext">Only Characters A-Z</p>
+      "! Only characters A-Z
+      only_chars_capital TYPE ty_simple_field_check VALUE 'A',
+      "! <p class="shorttext">Only Characters a-z</p>
+      "! Only characters a-z
+      only_chars_lower   TYPE ty_simple_field_check VALUE 'a',
+      "! <p class="shorttext">Only Characters A-Z, a-z</p>
+      "! Only characters A-Z, a-z
+      only_chars         TYPE ty_simple_field_check VALUE 'Z',
+      "! <p class="shorttext">Alphanumeric with special chars and German Umlauts</p>
+      "! Alphanumeric with special chars and German Umlauts
+      x                  TYPE ty_simple_field_check VALUE 'X',
+    END OF co_simple_field_check.
+  CONSTANTS:
+    BEGIN OF co_operator,
+      "! <p class="shorttext">Equal</p>
+      "! Equal
+      equal                      TYPE ty_operator VALUE 'E',
+      "! <p class="shorttext">Non Equal</p>
+      "! Non equal
+      non_equal                  TYPE ty_operator VALUE 'N',
+      "! <p class="shorttext">Empty or '0'</p>
+      "! Empty or '0'
+      contains_pattern           TYPE ty_operator VALUE 'CP',
+      "! <p class="shorttext">No Pattern</p>
+      "! No pattern
+      no_pattern                 TYPE ty_operator VALUE 'NP',
+      "! <p class="shorttext">Contains Any Character</p>
+      "! Contains any character
+      contains_any_character     TYPE ty_operator VALUE 'CA',
+      "! <p class="shorttext">Not Contains Any Character</p>
+      "! Not contains any character
+      not_contains_any_character TYPE ty_operator VALUE 'NA',
+      "! <p class="shorttext">Matches (regular expression)</p>
+      "! Matches (regular expression)
+      matches                    TYPE ty_operator VALUE 'MT',
+    END OF co_operator.
+  CONSTANTS:
+    BEGIN OF co_operator_db,
```

Comment:
in interface zif_aff_types_v1 we have constant co_option.
Do you want to use the constants from zif_aff_types_v1=>co_option?
But for "equal" and "non_equal" the values are different than yours.